spi/src/main/java/com/flipkart/security/grayskull/spi/EncryptionService.java

```diff
@@ -0,0 +1,6 @@
+package com.flipkart.security.grayskull.spi;
+
+public interface EncryptionService {
+    byte[] encrypt(byte[] data, String keyId);
+    byte[] decrypt(byte[] data, String keyId);
```

Comment:
might not be needed for this PR, but we can think of adding two more methods here 
1. getKeyMetadata : to identify if the encryption key has been rotated on the KMS side. helpful in rotation
2. rewrap : decrypt and encrypt data with the latest version of the encryption key.